libcst/codemod/_cli.py

```diff
@@ -607,13 +609,20 @@ def parallel_exec_transform_with_prettyprint(  # noqa: C901
         python_version=python_version,
     )
 
+    pool_impl: Callable[[], Executor]
     if total == 1 or jobs == 1:
         # Simple case, we should not pay for process overhead.
-        # Let's just use a dummy synchronous pool.
+        # Let's just use a dummy synchronous executor.
         jobs = 1
-        pool_impl = DummyPool
+        pool_impl = DummyExecutor
+    elif getattr(sys, "_is_gil_enabled", lambda: False)():
```

Comment:
```suggestion
    elif not getattr(sys, "_is_gil_enabled", lambda: True)():
```

The way you have this written now it'll use a process pool if the GIL is disabled.

That said, it looks like there are some scaling bottlenecks:

```
PYTHON_GIL=1 python3 -m libcst.tool codemod --no-format  .  307.24s user 3.77s system 979% cpu 31.750 total
```

vs

```
PYTHON_GIL=0 python3 -m libcst.tool codemod --no-format  .  561.45s user 1195.46s system 820% cpu 3:34.01 total
```

Reply:
(this is using the format-to-fstring codemod on the numpy repo - I'm gonna make a NumPy PR for that cleanup!)

Reply:
> The way you have this written now it'll use a process pool if the GIL is disabled.

That's the intention. We should only use threadpool if we do have a `sys._is_gil_enabled` function and it returns true. Otherwise fall back to ProcessPool which will be much faster

Reply:
I’m confused, why do you expect a process pool to be faster than a thread pool when the GIL is disabled?

Reply:
Boolean algebra is my nemesis, that's why

Reply:
You aren’t the first and won’t be the last to be tripped up by how they decided to spell that in the `sys` API.

Reply:
Oh and it's not even my first trip-up! 😆 I consistently mess this up